tests/test_type_coverage_generation.py

```diff
@@ -227,3 +232,141 @@ class OptionalIntFactory(DataclassFactory[OptionalInt]):
 
     assert isinstance(results[0].i, int)
     assert results[1].i is None
+
+
+def type_exists_at_path_any(objs: Iterable, path: List[Union[int, str]], target_type: Type) -> bool:
+    return any(type_exists_at_path(obj, path, target_type) for obj in objs)
+
+
+def type_exists_at_path(obj: Any, path: List[Union[int, str]], target_type: Type) -> bool:
```

Comment:
This is for checking the coverage rather than the validity of the object. This function is used by the function above to check a set of objects representing coverage of some parent actually has a particular permutation.

Reply:
Would it be possible to test the concrete variable instead or maybe using `dirty-equal` for this util? I think it'd be preferable to not maintain specific utils for this if possible

Reply:
> Commit history and state seems to be out of sync with main. Can you sync up with main please?

rebased

> Would it be possible to test the concrete variable instead or maybe using dirty-equal for this util?

Not really. It seems to me that the inclusion dirty-equal would just be replacing an isinstance check with extra steps. 

> I think it'd be preferable to not maintain specific utils for this if possible

The specific utility here is purely for the tests in this file. It kind of __is__ the main testing mechanism for coverage. I think it is necessary and I don't think there is much danger of having to maintain it for other use cases.

Reply:
> Would it be possible to test the concrete variable instead or maybe using dirty-equal for this util?

Not really. It seems to me that the inclusion dirty-equal would just be replacing an isinstance check with extra steps. 

> I think it'd be preferable to not maintain specific utils for this if possible

The specific utility here is purely for the tests in this file. It kind of __is__ the main testing mechanism for coverage. I think it is necessary and I don't think there is much danger of having to maintain it for other use cases.

Reply:
I think we can revisit this at a later point. For now, I think this is fine.